src/System.Private.CoreLib/src/System/Threading/Thread.CoreRT.Unix.cs

```diff
@@ -55,6 +55,10 @@ private static void OnThreadExit()
                 {
                     currentThread.SetThreadStateBit(ThreadState.Stopped);
                 }
+                if ((state & (int)ThreadState.Background) == 0)
+                {
+                    DecrementRunningForeground();
```

Comment:
@jkotas I've moved bookkeeping here from `StartThread`'s finally block. But we have such ability to register thread shutdown callback only for Unix at the moment.
I've searched the internets and it appears there is such [opportunity](https://docs.microsoft.com/en-us/windows/win32/api/winnt/nc-winnt-pfls_callback_function) with Windows fibers.
Seems this is exactly what we need https://github.com/dotnet/corert/blob/177127827a3716e41b4019fdc2a1d430cec43e54/src/Native/Runtime/windows/PalRedhawkMinWin.cpp#L58-L68 
Three calls down and we are in `PalDetachThread` which on Unix PAL calls our `OnThreadExit` while on Windows PAL this just clears fiber's thread. So I guess we could register callback similar to Unix way and call it from Window's `PalDetachThread`.

Reply:
Agree.

I think it is what we need. I am wondering whether we are going to see problems caused by managed code running during thread detach. We can deal with that problem once we hit it.

Reply:
FYI: Windows agents will hung until this implemented (or cleanup moved to some pal-independent place). I hope there are timeouts configured on agents.